database/sql/src/main/scala/cromwell/database/slick/SlickDatabase.scala

```diff
@@ -159,10 +160,30 @@ abstract class SlickDatabase(override val originalDatabaseConfig: Config) extend
     database.close()
   }
 
+  private val debugExitConfigPath = "danger.debug.only.exit-on-rollback-exception-with-status-code"
+  private val debugExitStatusCodeOption = ConfigFactory.load.getAs[Int](debugExitConfigPath)
+
   protected[this] def runTransaction[R](action: DBIO[R], isolationLevel: TransactionIsolation = TransactionIsolation.RepeatableRead): Future[R] = {
-    //database.run(action.transactionally) <-- See comment above private val actionThreadPool
-    val dbio = action.transactionally.withTransactionIsolation(isolationLevel)
-    Future(Await.result(database.run(dbio), Duration.Inf))(actionExecutionContext)
+    runAutoCommit(action.transactionally.withTransactionIsolation(isolationLevel))
+  }
+
+  protected[this] def runAutoCommit[R](action: DBIO[R]): Future[R] = {
+    //database.run(action) <-- See comment above private val actionThreadPool
+    Future {
+      try {
+        Await.result(database.run(action), Duration.Inf)
```

Comment:
Is this functionally different from a Future with a recover for `MySQLTransactionRollbackException`?

Reply:
Yes, it's _slightly_ different.

`try/catch` doesn't create a new `Future`, while using [`recover`](https://www.scala-lang.org/api/2.12.6/scala/concurrent/Future.html) would. `recover` requires plumbing through an execution context to create its new `Future`. That non-database EC is currently not plumbed through (but easily could be). Plus IMHO this java-esque hack is working well enough around the already existing hacks.